blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanDomainSpecification.java

```diff
@@ -0,0 +1,26 @@
+package io.jenkins.blueocean.rest.impl.pipeline.credential;
+
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.cloudbees.plugins.credentials.domains.DomainSpecification;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This is BlueOcean specific {@link DomainSpecification}.
+ *
+ * @author Vivek Pandey
+ */
+public class BlueOceanDomainSpecification extends DomainSpecification {
+    public static final String DOMAIN_SPECIFICATION = "BlueOcean Credentials Domain Specification";
+
+    @Nonnull
+    @Override
+    public Result test(@Nonnull DomainRequirement scope) {
+        if(scope instanceof BlueOceanDomainRequirement){
+            if(((BlueOceanDomainRequirement)scope).getValue().equals(DOMAIN_SPECIFICATION)){
+                return Result.POSITIVE;
+            }
+        }
+        return Result.NEGATIVE;
```

Comment:
FYI by returning  NEGATIVE we prevent the credential from being visible to the user until they are searching with the `BlueOceanDomainRequirement`. In other words, `BlueOceanDomainRequirement` is *MUST HAVE*.

If you want the user to be able to use the credential in other contexts without the context being aware of your requirement then you may want to return `UNKNOWN`.